metadata-ingestion/src/datahub/ingestion/source/aws/aws_common.py

```diff
@@ -64,6 +66,25 @@ def assume_role(
 AUTODETECT_CREDENTIALS_DOC_LINK = "Can be auto-detected, see [the AWS boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html) for details."
 
 
+class LazyLogEvaluator:
+    """
+    Used by debug logging to avoid costly function calls if the information is not logged (due to the logging level set
+    higher than DEBUG)
+    """
+
+    def __init__(self, callback, *args):
+        self.callback = callback
+        self.args = args
+        self.value_stored = False
+        self.value = None
+
+    def __repr__(self):
+        if not self.value_stored:
+            self.value = self.callback(*self.args)
+            self.value_stored = True
```

Comment:
Is this store mechanism used anywhere?

Reply:
Maybe I missed it, but you are not reusing this anywhere.

Reply:
It is used by `DEBUG` logger to avoid costly call to sts if we are not on `DEBUG` level. See here:
https://github.com/datahub-project/datahub/pull/11252/files#diff-658ffa764c667e22fae4a46946899527226fa9e0ce7464d35cf56bd4c2a29726R220

Reply:
Yes, it was more because I saw you cache the value in LazyLogEvalator, but the LazyLogEvaulator instance is never reused.
It is not a problem as it can be useful in the future.

Reply:
It is reused actually. We create it once when log.debug line is hit. Then the __repr__ function is called (assuming DEBUG-level is set) to print to console. And then it is called again (now using the cached value) because beside logging to console we also log to a file if debugging is turned on, so caching is, in fact, used.